internal/security/fileioperformer/mocks/mock_test.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
```

Comment:
Not sure we need to test this.... mock's functionality should already be tested by 3rd party tool `Mock.mock`, no?  The reason I put here is that, it introduces the extra efforts for this on the maintenance efforts if somehow the interface changes.

Reply:
On the mocking objects: they exist so that I only need to write them once, rather than writing mocks in each dependent component (this particular one is used 4-5 times).  Most of the edgex components have a mocks folder with the mocks in them.

The mocks need to be tested to ensure that code coverage remains high.  Also, there actually is something substantial to the mock in that it needs to make sure the correct args are passed to .Called() and the correct arguments are returned.  I fact, if the interface changes, we do need a test that fails the mock.

`mock` doesn't actually test anything.  It just provides a framework to memoize the results of calling functions with certain arguments.  Because golang doesn't have runtime code generation it is necessary to actually code the mock in a way that satisfies the type checker.

Reply:
ok